test/e2e/webhook_test.go

```diff
@@ -499,19 +505,25 @@ var _ = Describe("Fleet's CRD Resource Handler webhook tests", func() {
 		})
 
 		It("should deny UPDATE operation on Fleet CRD for user not in system:masters group", func() {
-			var crd v1.CustomResourceDefinition
-			Expect(HubCluster.KubeClient.Get(ctx, types.NamespacedName{Name: "memberclusters.fleet.azure.com"}, &crd)).Should(Succeed())
+			Eventually(func() error {
+				var crd v1.CustomResourceDefinition
+				Expect(HubCluster.KubeClient.Get(ctx, types.NamespacedName{Name: "memberclusters.fleet.azure.com"}, &crd)).Should(Succeed())
```

Comment:
Hi Arvind! If it's in an eventually block, you should return the error rather than setting expectations.

Reply:
Or use a passed in gomega object.

Reply:
Same for the other Expect blocks below.